src/MultiRunner.php

```diff
@@ -84,18 +89,27 @@ public function remove(PromiseCore $core): void
     public function wait(PromiseCore $targetCore = null): void
     {
         do {
-            $status = curl_multi_exec($this->multiHandle, $active);
+            if (curl_multi_select($this->multiHandle, self::SELECT_TIMEOUT) === -1) {
+                // See https://bugs.php.net/bug.php?id=61141
+                usleep(250);
+            }
+
+            do {
+                $status = curl_multi_exec($this->multiHandle, $active);
+                // TODO CURLM_CALL_MULTI_PERFORM never returned since cURL 7.20.
+            } while ($status === CURLM_CALL_MULTI_PERFORM);
+
             $info = curl_multi_info_read($this->multiHandle);
-            if (false !== $info) {
+            if ($info !== false) {
```

Comment:
please don't revert yoda style comparisons.

Reply:
Why?

Reply:
because it has nothing to do with the change at hand.

i would have expected the code style checker to complain, but seems we don't have a rule for this activated.